### PR TITLE
kfilt: init at 0.0.8

### PR DESCRIPTION
**Pull request by ryane**
###### Description of changes

kfilt can filter Kubernetes resources:
https://github.com/ryane/kfilt

###### Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [x] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [x] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [23.11 Release Notes (or backporting 23.05 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2305-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

File: pkgs/applications/networking/cluster/kfilt/default.nix [hunk 1]
```diff
@@ -0,0 +1,31 @@
...
+
+  vendorHash = "sha256-c77CzpE9cPyobt87uO0QlkKD+xC/tM7wOy4orM62tnI=";
+
+  subPackages = [ "." ];
+
+  ldflags = [
+    "-s"
+    "-w"
+    "-X github.com/ryane/kfilt/cmd.Version=${version}"
+    "-X github.com/ryane/kfilt/cmd.GitCommit=${version}"
```
Comment (emilylange) [inline pkgs/applications/networking/cluster/kfilt/default.nix:22]:
```suggestion
    "-X github.com/ryane/kfilt/cmd.GitCommit=${src.rev}"
```

Nit

File: pkgs/applications/networking/cluster/kfilt/default.nix [hunk 2]
```diff
@@ -0,0 +1,31 @@
+{ lib, stdenv, buildGo120Module, fetchFromGitHub }:
+
+buildGo120Module rec {
```
Comment (emilylange) [inline pkgs/applications/networking/cluster/kfilt/default.nix:3]:
```suggestion
buildGoModule rec {
```

`buildGoModule` already points to `1.20` and should be preferred.

If this package happens to fail to build with some new go version `buildGoModule` then points to, another maintainer temporarily overwrites this in the `all-packages.nix` instead :)

File: pkgs/applications/networking/cluster/kfilt/default.nix [hunk 3]
```diff
@@ -0,0 +1,31 @@
+{ lib, stdenv, buildGo120Module, fetchFromGitHub }:
```
Comment (emilylange) [inline pkgs/applications/networking/cluster/kfilt/default.nix:1]:
```suggestion
{ lib, buildGoModule, fetchFromGitHub }:
```

`stdenv` isn't used + suggestion below/above regarding `buildGoModule`

**Review by emilylange — CHANGES_REQUESTED** [review]

**Comment by ryane** [comment]
thanks for the feedback! I committed your suggestions.

**Comment by emilylange** [comment]
Great, thank you! :)

Please squash your two commits into a single one.
See https://github.com/NixOS/nixpkgs/blob/a30520bf8eabf8a5c37889d661e67a2dbcaa59e6/CONTRIBUTING.md#writing-good-commit-messages

File: maintainers/maintainer-list.nix [hunk 4]
```diff
@@ -13982,6 +13982,15 @@
     githubId = 889991;
     name = "Ryan Artecona";
   };
+  ryane = {
```
Comment (AndersonTorres) [inline maintainers/maintainer-list.nix:13985]: Split this in a separate commit.

There is no need to do a new pull request.

Repeating: 

**There is no need to do a new pull request.**

File: pkgs/applications/networking/cluster/kfilt/default.nix [hunk 5]
```diff
@@ -0,0 +1,31 @@
...
+    "-X github.com/ryane/kfilt/cmd.Version=${version}"
+    "-X github.com/ryane/kfilt/cmd.GitCommit=${src.rev}"
+  ];
+
+  meta = with lib; {
+    description = "Command-line tool that filters Kubernetes resources";
+    homepage = "https://github.com/ryane/kfilt";
+    license = licenses.asl20;
+    maintainers = [ maintainers.ryane ];
+  };
```
Comment (AndersonTorres) [inline pkgs/applications/networking/cluster/kfilt/default.nix:30]:
```suggestion
  meta = {
    description = "Command-line tool that filters Kubernetes resources";
    homepage = "https://github.com/ryane/kfilt";
    license = lib.licenses.asl20;
    maintainers = [ lib.maintainers.ryane ];
  };
```

**Review by AndersonTorres — CHANGES_REQUESTED** [review]

File: maintainers/maintainer-list.nix (hunk 4 above)
Comment (ryane) [reply maintainers/maintainer-list.nix:13985]: 985558701d98a1ac3c8ba4288b1764add5c93dda

**Review by AndersonTorres — APPROVED** [review]
Ping me 3 days and I will merge this.

**Comment by ryane** [comment]
> Ping me 3 days and I will merge this.

ping @AndersonTorres
